### PR TITLE
add basic authentication if https certificates are not used

### PR DESCRIPTION
**Pull request by arthurlogilab**

### CODE REVIEW

File: bin/foreman-node [hunk 1]
```diff
@@ -146,6 +146,10 @@ def upload_grains(minion)
         res.cert = OpenSSL::X509::Certificate.new(File.read(SETTINGS[:ssl_cert]))
         res.key  = OpenSSL::PKey::RSA.new(File.read(SETTINGS[:ssl_key]), nil)
       end
+    else
+      if SETTINGS[:username] && SETTING[:password]
```
Comment (stbenjam) [inline bin/foreman-node]: This can be `elsif` instead of spread across two lines please

File: bin/foreman-node [hunk 2]
```diff
@@ -170,6 +174,10 @@ def enc(minion)
       http.cert = OpenSSL::X509::Certificate.new(File.read(SETTINGS[:ssl_cert]))
       http.key  = OpenSSL::PKey::RSA.new(File.read(SETTINGS[:ssl_key]), nil)
     end
+  else
+    if SETTINGS[:username] && SETTING[:password]
```
Comment (stbenjam) [inline bin/foreman-node]: Same thing here

**Comment by stbenjam** [comment]
Thanks for the PR's! 

Just 2 little things, if you wouldn't mind fixing them, and I'll get both the PR's merged and in the next release.

File: bin/foreman-node (hunk 1 above)
Comment (arthurlogilab) [inline bin/foreman-node]: done

File: bin/foreman-node (hunk 2 above)
Comment (arthurlogilab) [inline bin/foreman-node]: done

**Comment by stbenjam** [comment]
@arthurlogilab Sorry, it's `elsif` (with an s) not `elif` in ruby

**Comment by arthurlogilab** [comment]
fixed

**Comment by mmoll** [comment]
merged, merci @arthurlogilab!

**Pull request merged**